### ISSUE TITLE
Help with V2 MultipartUpload

### ISSUE THREAD

**Issue by KateWilkins** [aws/aws-sdk-ruby#1377]
In V1 uploading was as simple as mpu.add_part().

How are parts added to the MPU object in V2, I can't find any MPU examples for V2.

I can manually create a MulitpartUploadPart.new with the right params and add it but that seems to be pointless given I have an initialised MultpartUpload object for the bucket.

Thanks
Kate

**Comment by awood45** [comment]
V2 approaches multipart uploads differently, so there are a couple of options.

One is to use the [managed file upload helper](http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Object.html#upload_file-instance_method).

The other is to use the relevant client operations directly:
* [#create_multipart_upload](http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#create_multipart_upload-instance_method)
* [#upload_part](http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#upload_part-instance_method)
    * [#upload_part_copy]() if using an existing S3 object as the source
* [#complete_multipart_upload](http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#complete_multipart_upload-instance_method)
* [#abort_multipart_upload](http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#abort_multipart_upload-instance_method) if needed

I'd generally recommend using the managed file uploader, unless there's a specific reason it doesn't work for your use case. If it doesn't work for your use case, do let me know if there's a feature we can add that would help make multipart uploads easier for you.

**Comment by KateWilkins** [comment]
Hi Alex,

Many thanks for the response, I cant use the managed file uploader as the uploaded contents are dynamically generated in memory, I could flush to disk but it seems a waste and more convoluted.

I've re-written the code to use the client operations as suggested above and its working fine.

It is a little puzzling as to why there aren't any worked examples of how to use the V2 MultipartUpload & MultipartUploadPart objects or some directions in the API docs to say as above and use the client operations.

I googled "aws ruby multipart upload" and got this link

http://docs.aws.amazon.com/AmazonS3/latest/dev/uploadobjusingmpu-ruby-sdk.html

it pushes developers towards the MPU objects (V1) without any examples then of course I clicked to the V2 version and there I got stuck..... It might be worth updating it with the what you have above as that was all I needed to get it working.

Thanks again.
Kate
